server/src/main/java/org/elasticsearch/index/IndexingPressure.java

```diff
@@ -193,7 +204,49 @@ public Releasable markPrimaryOperationLocalToCoordinatingNodeStarted(int operati
         });
     }
 
-    public Releasable markPrimaryOperationStarted(int operations, long bytes, boolean forceExecution) {
+    public void checkLargestPrimaryOperationIsWithinLimits(
```

Comment:
Any reason this should not be called from within `markPrimaryOperationLocalToCoordinatingNodeStarted` similar to `validateAndMarkPrimaryOperationStarted`?

Reply:
Not really, I added that method in https://github.com/elastic/elasticsearch/pull/123543/commits/f8b88141892d3ca9eb16470a177c6b46191fa377